hs_core/models.py

```diff
@@ -1349,14 +1363,29 @@ class Meta:
         unique_together = ("content_type", "object_id")
 
 def get_path(instance, filename):
-    return os.path.join(instance.content_object.short_id, 'data', 'contents', filename)
+    if instance.content_object.resource_federation_path:
+        return os.path.join(instance.content_object.resource_federation_path, instance.content_object.short_id, 'data',
+                            'contents', filename)
+    else:
+        return os.path.join(instance.content_object.short_id, 'data', 'contents', filename)
 
 class ResourceFile(models.Model):
```

Comment:
I was thinking if adding functions like the followings to this class would abstract away from the rest of the code base the need to know which iRODS zone the file exists:
```copy_file(self, params)```
```move_files(self, params)```
```get_file(self, params)```
......